doc/otn/OTAI-v0.0.1-specification.md

```diff
@@ -0,0 +1,909 @@
+| [APS Port](#automatic-protection-switch-port-functionality-otaiapsporth) | Automatic Protection Switch Port | [otaiapsport.h](https://github.com/zhengweitang-zwt/OTAI/blob/main/inc/otaiapsport.h) |
+
+Here is OTAI objects and hierarchy, the top-level object is the linecard object, which could be a real hardware on multi linecard device, or a logical system on a pizza box. The linecard may contains a set of optical components and logical objects. For instance, an optical transponder and muxponder linecard includes Ports(client and line port), transceivers, and logical objects (logical channels, physical channels, ethernet, OTN, OCH, LLDP, etc). An optical linecard in the line system, it may includes OA, OSC, APS, Attenuators, WSS, Media Channel, OCM, OTDR, Ports, etc.  
+
+<img src="../../images/otn/OTAI-objects-hierarchy.png" alt="OTAI object model and hierarchy" style="zoom: 40%;" />
+
+Each OTAI objects owns a unique object ID (OID) which is assigned by OTAI library. Each OTAI object has a set of attributes and statistics which are compatible with OpenConfig definitions, and a set of CRUD (Create/Read/Update/Delete) APIs. With these standard OTAI APIs and unique OID, the adapter host can manage all kinds of optical transport components from different vendors.
+
+## Adapter startup/shutdown sequence  
+Adapter Host has to load the information about OTAI adapter module during startup. After adapter module load, adapter host acquire three well-known functions addresses: otai_api_initialize(), otai_api_query() and otai_api_uninitialize().  
+
+After the otai_api_initialize() is called, the function is supplied with method table of services provided by the adapter host to adapter module. These services allow the adapter to initialize any data, control structures that may be necessary during subsequent OTAI operations. Note: SDK initialization should NOT be performed here. This function is just take care of any platform-specific difference related to module loading, and generally should be very simple.  
+
+After initialization, otai_api_query()can be used for retrieval of various methods tables for OTAI functionalities. Once all the functionalities being retrieved, adapter host has all OTAI objects CRUD APIs.
+For example, the first method that Adapter hosts is `otai_create_linecard_fn` which create the top otai object and return the linecard object ID to the Adapter host. Then Adapter host can create all the other sub-components (Ports, OAs, OLP, VOA, etc) in this linecard with the same linecard object ID. 
+Once the linecard and its sub-components are created, Adapter host be able to retrieve all the statistic and status with the API `otai_get_**_attribute_fn` and `otai_get_**_stats_fn`, set these optical components' attributes with the API `otai_set_**_attribute_fn`.  
+
+## Data Types (otaitypes.h)  
```

Comment:
is it possible to reuse saitypes.h, which defined most of these?

Reply:
It's difficult to link files from another repository on the github. User should be able to compile and test OTAI without download other repository's source code.